app/helpers/planning_application_helper.rb

```diff
@@ -27,12 +27,48 @@ def filter_text
     end
   end
 
-  def map_link(address)
-    "https://google.co.uk/maps/place/#{CGI.escape(address)}"
+  def flooding_constraints
+    ["Flood zone", "Flood zone 1", "Flood zone 2", "Flood zone 3"]
+  end
+
+  def military_constraints
+    ["Explosives & ordnance storage", "Safeguarded land"]
+  end
+
+  def ecology_constraints
+    ["Special Area of Conservation (SAC)", "Site of Special Scientific Interest (SSSI)", "Ancient Semi-Natural Woodland (ASNW)", "Local Wildlife / Biological notification site", "Priority habitat"]
+  end
+
+  def heritage_constraints
+    ["Listed building", "Conservation Area", "Area of Outstanding Natural Beauty", "National Park", "World Heritage Site", "Broads"]
+  end
+
+  def policy_constraints
+    ["Article 4 area", "Green belt"]
+  end
+
+  def tree_constraints
+    ["Tree Preservation Order"]
+  end
+
+  def other_constraints
+    ["Safety hazard area", "Within 3km of the perimeter of an aerodrome"]
   end
 
-  def list_constraints(constraints)
-    JSON.parse(constraints).select { |_category, value| value == true }.keys unless constraints.empty?
+  def existing_constraints(constraints)
+    constraints - standard_constraints
+  end
+
+  def standard_constraints
+    [flooding_constraints, military_constraints, ecology_constraints, heritage_constraints, policy_constraints, tree_constraints, other_constraints].flatten
+  end
+
+  def check_added?(constraint, existing_constraints)
+    JSON.parse(existing_constraints).include?(constraint) ? true : false
```

Comment:
Again, not sure on the naming of this method, not entirely clear what it is for. And does it need JSON.parse? I thought we'd got rid of JSON?

Reply:
We can get rid of this method - it was left over from before. The reason it's not throwing an error is because it's not used anywhere

Reply:
Removed.